Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,369 @@
+    {f : ι → α → β} (hf : ∀ i, Memℒp (f i) p μ) : UnifTight f p μ := fun ε hε ↦ by
+  by_cases hε_top : ε = ∞
+  · exact ⟨∅, by measurability, fun _ => hε_top.symm ▸ le_top⟩
+  by_cases hι : Nonempty ι
+  case neg => exact ⟨∅, (by measurability), fun i => False.elim <| hι <| Nonempty.intro i⟩
+  cases' hι with i
+  obtain ⟨s, _, hμs, hfε⟩ := (hf i).exists_snorm_indicator_compl_lt hp_top hε.ne'
+  refine ⟨s, ne_of_lt hμs, fun j => ?_⟩
+  convert hfε.le
+
+/-- This lemma is less general than `MeasureTheory.unifTight_finite` which applies to
+all sequences indexed by a finite type. -/
+private theorem unifTight_fin (hp_top : p ≠ ∞) {n : ℕ} {f : Fin n → α → β}
+    (hf : ∀ i, Memℒp (f i) p μ) : UnifTight f p μ := by
+  revert f
+  induction' n with n h
+  · intro f hf
+    have : Subsingleton (Fin Nat.zero) := subsingleton_fin_zero -- Porting note: Added this instance
+    exact unifTight_of_subsingleton hp_top hf
+  intro f hfLp ε hε
+  by_cases hε_top : ε = ∞
+  · exact ⟨∅, (by measurability), fun _ => hε_top.symm ▸ le_top⟩
+  let g : Fin n → α → β := fun k => f k
+  have hgLp : ∀ i, Memℒp (g i) p μ := fun i => hfLp i
+  obtain ⟨S, hμS, hFε⟩ := h hgLp hε
+  obtain ⟨s, _, hμs, hfε⟩ := (hfLp n).exists_snorm_indicator_compl_lt hp_top hε.ne'
+  refine ⟨s ∪ S, (by measurability), fun i => ?_⟩
+  by_cases hi : i.val < n
+  · rw [(_ : f i = g ⟨i.val, hi⟩)]
+    · rw [compl_union, ← indicator_indicator]
+      apply (snorm_indicator_le _).trans
+      exact hFε (Fin.castLT i hi)
+    · simp only [Fin.coe_eq_castSucc, Fin.castSucc_mk, g]
+  · rw [(_ : i = n)]
+    · rw [compl_union, inter_comm, ← indicator_indicator]
+      apply (snorm_indicator_le _).trans
+      convert hfε.le
+    · have hi' := Fin.is_lt i
+      rw [Nat.lt_succ_iff] at hi'
+      rw [not_lt] at hi
+      -- Porting note: Original proof was `simp [← le_antisymm hi' hi]`
+      ext; symm; rw [Fin.coe_ofNat_eq_mod, le_antisymm hi' hi, Nat.mod_succ_eq_iff_lt, Nat.lt_succ]
+
+/-- A finite sequence of Lp functions is uniformly tight. -/
+theorem unifTight_finite [Finite ι] (hp_top : p ≠ ∞) {f : ι → α → β}
+    (hf : ∀ i, Memℒp (f i) p μ) : UnifTight f p μ := fun ε hε ↦ by
+  obtain ⟨n, hn⟩ := Finite.exists_equiv_fin ι
+  set g : Fin n → α → β := f ∘ hn.some.symm
+  have hg : ∀ i, Memℒp (g i) p μ := fun _ => hf _
+  obtain ⟨s, hμs, hfε⟩ := unifTight_fin hp_top hg hε
+  refine ⟨s, hμs, fun i => ?_⟩
+  specialize hfε (hn.some i)
+  unfold_let g at hfε
+  simp_rw [Function.comp_apply, Equiv.symm_apply_apply] at hfε
+  assumption
+
+end UnifTight
+
+
+section VitaliConvergence
+
+variable {μ : Measure α} {p : ℝ≥0∞}
+
+variable {f : ℕ → α → β} {g : α → β}
+
+/-! Both directions and an iff version of Vitali's convergence theorem on measure spaces
+   of not necesserily finite volume. See `Thm III.6.15` of Dunford & Schwartz, Part I (1958). -/
+
+/- We start with the reverse direction. We only need to show that uniform tightness follows
```

Comment:
Hmm, I see this more of an internal comment about the implementation, rather than a public facing doc-string. Same for the later "/- Next we deal with ... -/" comment.

Reply:
Sure. The point about not indenting still stands, though